src/compositions/commons/Modal.tsx

```diff
@@ -13,19 +14,35 @@ interface ModalProps {
 
 export const Modal: VFC<ModalProps> = (props: ModalProps) => {
   const [isBrowser, setIsBrowser] = useState(false)
+  const modal = useRef<HTMLDivElement>(null)
 
   useEffect(() => {
     setIsBrowser(true)
   }, [])
 
+  useEffect(() => {
+    if (!props.show) return
+
+    function closeModalOnClickAway(event: MouseEvent) {
+      const eventTargetElement = event.target as HTMLElement
+      if (modal.current && !modal.current.contains(eventTargetElement)) {
+        props.onClose()
+      }
+    }
+
+    window.addEventListener('click', closeModalOnClickAway)
```

Comment:
- bylo sposob nie uzywajac "window.addEventListener" chybaaaa, kiedys implementowalem
- jak uzywasz **string ('click')**, to osobiscie declare gdzies const najpierw (sposob Mfro, Mazrica)
```
// tez powinno byc w innym pliku, ale suzuchi ginkou nie jest duza aplikacja, wiec hmmm....
const EVENT_CLICK = 'click'
```
- pisanie bezposrednio jak 'click', to "hard coding". Mowilem ci w tym tygodniu desu. O to chodzi